nameko/messaging.py

```diff
@@ -75,22 +74,22 @@ def unpack_message_headers(self, worker_ctx_cls, message):
 
 class Publisher(DependencyProvider, HeaderEncoder):
 
-    def __init__(self, exchange=None, queue=None):
+    def __init__(self, exchange=None, queues_to_declare=None):
```

Comment:
It would be nice to continue to accept the `queue` argument for backwards compatibility.

If you pass `exchange=None` and `queue=<some bound queue>` to the current implementation, it'll declare the queue and publish to the exchange the queue is bound to. So to maintain compatibility you'd need to:

* Accept a queue argument
* If the queue is bound and no exchange is provided, set `self.exchange` to the queue's exchange
* Add the queue to `queues_to_declare`
* Ideally raise a warning noting that the `queue` arg is deprecated